src/test/java/com/nimbusds/jose/JWEObjectJSONTest.java

```diff
@@ -0,0 +1,354 @@
+                    agreementPartyUInfo(Base64URL.encode("Alice")).
+                    agreementPartyVInfo(Base64URL.encode("Bob, Charlie")).
+                    build();
+
+            JWEObjectJSON jweObject = new JWEObjectJSON(header, payload);
+
+            ECDHEncrypterMulti encrypter = new ECDHEncrypterMulti(new ECKey[]{ aliceKey, bobKey, charlieKey });
+            encrypter.getJCAContext().setContentEncryptionProvider(BouncyCastleProviderSingleton.getInstance());
+            jweObject.encrypt(encrypter);
+
+            ECKey epk = (ECKey) jweObject.getHeader().getEphemeralPublicKey();
+            assertEquals(cycle.curve, epk.getCurve());
+            assertNotNull(epk.getX());
+            assertNull(epk.getD());
+
+            String jwe = jweObject.serialize();
+            jweObject = JWEObjectJSON.parse(jwe);
+            assertNotNull(jweObject.getRecipients());
+            assertEquals(3, jweObject.getRecipients().size());
+
+            ECDHDecrypterMulti decrypter = new ECDHDecrypterMulti(new ECKey[] { aliceKey, bobKey, charlieKey });
+            decrypter.getJCAContext().setContentEncryptionProvider(BouncyCastleProviderSingleton.getInstance());
+            jweObject.decrypt(decrypter);
+
+            assertEquals(payload.toString(), jweObject.getPayload().toString());
+        }
+    }
+
+    public void test_forbiddenECDH_1PUX25519Cycles() throws Exception {
+        Payload payload = new Payload("Hello world!");
+
+        for (CycleTest cycle : forbiddenECDH_1PUX25519Cycles) {
+            OctetKeyPair aliceKey = generateOKP(cycle.curve, "1");
+            OctetKeyPair bobKey = generateOKP(cycle.curve, "2");
+            OctetKeyPair charlieKey = generateOKP(cycle.curve, "3");
+
+            JWEHeader header = new JWEHeader.Builder(cycle.algorithm, cycle.encryptionMethod).
+                    agreementPartyUInfo(Base64URL.encode("Alice")).
+                    agreementPartyVInfo(Base64URL.encode("Bob, Charlie")).
+                    build();
+
+            JWEObjectJSON jweObject = new JWEObjectJSON(header, payload);
+
+            try {
+                ECDH1PUX25519EncrypterMulti encrypter = new ECDH1PUX25519EncrypterMulti(aliceKey,
+                        new OctetKeyPair[]{ bobKey.toPublicJWK(), charlieKey.toPublicJWK() });
+
+                encrypter.getJCAContext().setContentEncryptionProvider(BouncyCastleProviderSingleton.getInstance());
+                jweObject.encrypt(encrypter);
+                fail();
+            } catch (JOSEException e) {
+                String expectedMessage = String.format("Unsupported JWE encryption method %s, " +
+                                "must be A128CBC-HS256, A192CBC-HS384 or A256CBC-HS512",
+                        cycle.encryptionMethod.getName());
+                assertEquals(expectedMessage, e.getMessage());
+            }
+        }
+    }
+
+    public void test_ecdh_1pu_a128kw_appendix() throws Exception {
```

Comment:
It looks like this is the test for decryption part only. Let's add the full example for encryption from Appendix B checking all test vectors from there.